torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -3071,32 +3073,40 @@ class ForceFP16Annotation : public IterVisitor {
         [](auto* val) {
           return val->template isA<TensorView>() &&
               val->getDataType().has_value() &&
-              val->getDataType().value() == DataType::Half;
+              (val->getDataType().value() == DataType::Half || val->getDataType().value()== DataType::BFloat16);
         });
 
     annotation.traverseFrom(fusion, fp16_outputs);
     return annotation.force_fp16_tv_set_;
   }
 
+
  private:
   using IterVisitor::handle;
 
   void handle(TensorView* tv) override {
     auto dtype = tv->getDataType();
     if (dtype.has_value() && dtype.value() == DataType::Float &&
-        !tv->isFusionOutput() && !tv->isFusionInput()) {
+        !tv->isFusionOutput() && !tv->isFusionInput() && (dtype == DataType::Half || dtype == DataType::BFloat16)) {
+      if (cast_to_type_){
```

Comment:
I think this wouldn't be best place to set cast_to_type because it's in a traversal that's identifying the tensors in fp32 that we need to cast to fp16. Suggested a place to change at line 3076.